keyboards/kyria/keymaps/jhelvy/keymap.c

```diff
@@ -0,0 +1,203 @@
+        0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  1,  1,  3,  7, 15, 14, 30, 28, 60, 56,120,112,112,112,224,224,224,224,224,224,224,224,224,224,224,224,224,224,224,224,112,112,112,120, 56, 60, 28, 30, 14, 15,  7,  3,  1,  1,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0
+    };
+    oled_write_raw_P(kyria_logo, sizeof(kyria_logo));
+}
+
+static void render_qmk_logo(void) {
+  static const char PROGMEM qmk_logo[] = {
+    0x80,0x81,0x82,0x83,0x84,0x85,0x86,0x87,0x88,0x89,0x8a,0x8b,0x8c,0x8d,0x8e,0x8f,0x90,0x91,0x92,0x93,0x94,
+    0xa0,0xa1,0xa2,0xa3,0xa4,0xa5,0xa6,0xa7,0xa8,0xa9,0xaa,0xab,0xac,0xad,0xae,0xaf,0xb0,0xb1,0xb2,0xb3,0xb4,
+    0xc0,0xc1,0xc2,0xc3,0xc4,0xc5,0xc6,0xc7,0xc8,0xc9,0xca,0xcb,0xcc,0xcd,0xce,0xcf,0xd0,0xd1,0xd2,0xd3,0xd4,0};
+
+  oled_write_P(qmk_logo, false);
+}
+
+static void render_status(void) {
+    // QMK Logo and version information
+    render_qmk_logo();
+    oled_write_P(PSTR("       Kyria rev1.0\n\n"), false);
+
+    // Host Keyboard Layer Status
+    oled_write_P(PSTR("Layer: "), false);
+    switch (get_highest_layer(layer_state)) {
+        case QWERTY:
+            oled_write_P(PSTR("QWERTY\n"), false);
+            break;
+        case HOTKEYS:
+            oled_write_P(PSTR("HOTKEYS\n"), false);
+            break;
+        case TEXT:
+            oled_write_P(PSTR("TEXT\n"), false);
+            break;
+        case SELECT:
+            oled_write_P(PSTR("SELECT\n"), false);
+            break;
+        case NUMBERS:
+            oled_write_P(PSTR("NUMBERS\n"), false);
+            break;
+        default:
+            oled_write_P(PSTR("Undefined\n"), false);
+    }
+
+}
+
+void oled_task_user(void) {
+    if (is_keyboard_master()) {
+        render_status(); // Renders the current keyboard state (layer, lock, caps, scroll, etc)
+    } else {
+        render_kyria_logo();
+    }
+}
+#endif
+
+void encoder_update_user(uint8_t index, bool clockwise) {
+    if (IS_LAYER_ON(HOTKEYS)) {
+        if (clockwise) {
+            tap_code(KC_VOLD);
+        } else {
+            tap_code(KC_VOLU);
+        }
+    } else if (IS_LAYER_ON(TEXT)) {
+        if (clockwise) {
+            tap_code(KC_MS_WH_LEFT);
+        } else {
+            tap_code(KC_MS_WH_RIGHT);
+        }
+    } else if (IS_LAYER_ON(NUMBERS)) {
+        if (clockwise) {
+            register_code(KC_LGUI);
+            tap_code(KC_MINS);
+            unregister_code(KC_LGUI);
+        } else {
+            register_code(KC_LGUI);
+            tap_code(KC_EQL);
+            unregister_code(KC_LGUI);
```

Comment:
```suggestion
            tap_code16(G(KC_EQL));
```

Reply:
what do these additions do?

Reply:
The same thing as the three lines you have, but in one line.

Reply:
Oh cool, I was trying to figure out what the "G()" part was, but I guess that's the LGUI hold.